runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -21,64 +24,77 @@
 @Slf4j
 public class ManagementView extends AbstractView {
 
-    private final long layoutRefreshTimeout = 500;
+    static final Duration WORKFLOW_TIMEOUT = Duration.ofMinutes(10);
+    static final Duration QUERY_INTERVAL = Duration.ofSeconds(1);
+    static final int WORKFLOW_RETRY = 3;
 
     public ManagementView(@NonNull CorfuRuntime runtime) {
         super(runtime);
     }
 
-    /**
-     * Checks if a given workflow with the given workflow ID is active or not.
-     * If returns True - the workflow is active
-     * If returns False - the workflow is either completed successfully or failed.
-     *
-     * @param server     Node endpoint on which the workflow is running.
-     * @param workflowId Workflow UUID
-     * @return True if active. Else False
-     */
-    public boolean isWorkflowActive(String server, UUID workflowId) {
-        return layoutHelper(l -> {
-            QueryResponse response = runtime.getRouter(server)
-                    .getClient(ManagementClient.class)
-                    .queryRequest(workflowId);
-            log.info("isWorkflowActive: {} on server:{} = {}",
-                workflowId, server, response.isActive());
-            return response.isActive();
-        });
+    void waitForWorkflow(UUID workflow, ManagementClient client) {
+        long tries = WORKFLOW_TIMEOUT.getSeconds() / QUERY_INTERVAL.getSeconds();
+        log.info("waitForWorkflow: num tries {}", tries);
+        for (long x = 0; x < tries; x++) {
+            try {
+                if (!client.queryRequest(workflow).isActive()) {
+                    break;
+                }
+            } catch (NetworkException e) {
+                log.warn("waitForWorkflow: Network exception while waiting for {} on try {}", workflow, x);
+            }
+            Sleep.sleepUninterruptibly(QUERY_INTERVAL);
+            log.info("waitForWorkflow: waiting for {} on try {}", workflow, x);
+        }
+    }
+
+    Layout getMaxLayout(CorfuRuntime rt) {
```

Comment:
Should be private? Also, what is it for? It probably doesn't do what you expect it to do

Reply:
Decrease the chance of the layout's epoch to regress and makes the operation less likely to retry.

Reply:
This seems to be a hack, maybe we should keep the layout from regressing in the first place instead.

Reply:
I'll move this to invalidate layout